### ISSUE TITLE
Proposal: task graph visualization

### ISSUE THREAD

**Issue by emlys** [natcap/taskgraph#94]
It's often helpful to have some flowchart-style visualization of data flow through a model. It would be great if taskgraph could generate a basic graph visualization. The `dependent_tasks` attribute should be enough to produce a dependency graph.

Something like [Graphviz](https://graphviz.org/) and its [python wrapper](https://pypi.org/project/graphviz/) could make this pretty straightfoward.

**Comment by phargogh** [comment]
I agree, even a basic visualization showing task dependencies and task names would be very helpful!  Maybe there would be a way to also show which files are being created from a given task?  Sounds useful, but I'm sure we wouldn't want the graphic to have so much information on it that we can't really parse it out.

Down the road, I could see a nice extension of this being interactivity, especially for very large graphs like we can find in some research scripts.  I had an issue with the NCI water quality pipeline where there were so many tasks that trying to understand the flow of dependencies in a Snakemake-generated Graphviz graphic (below) was pretty tough!  Interactivity that would allow one to see a path through the graph (e.g., mousing over a node and having its ancestor and dependent tasks and paths highlighted) would have been useful to me in NCI.
![dag](https://github.com/natcap/taskgraph/assets/4583897/0d59461b-0799-4eac-80b2-ade97344bb99)
